lib/src/widgets/mutation.dart

```diff
@@ -37,70 +42,61 @@ class Mutation extends StatefulWidget {
 class MutationState extends State<Mutation> {
   GraphQLClient client;
   ObservableQuery observableQuery;
-  StreamSubscription<QueryResult> onCompleteSubscription;
-
-  void runMutation(Map<String, dynamic> variables) {
-    observableQuery.setVariables(variables);
 
-    if (widget.onCompleted != null) {
-      onCompleteSubscription = observableQuery.stream.listen(
-        (QueryResult result) {
-          widget.onCompleted(result);
-          onCompleteSubscription.cancel();
-        },
+  WatchQueryOptions get _options => WatchQueryOptions(
+        document: widget.options.document,
+        variables: widget.options.variables,
+        fetchPolicy: widget.options.fetchPolicy,
+        errorPolicy: widget.options.errorPolicy,
+        fetchResults: false,
+        context: widget.options.context,
       );
-    }
 
-    observableQuery.controller.add(
-      QueryResult(
-        loading: true,
-      ),
-    );
+  ObservableQuery _replaceObservableQuery(Map<String, dynamic> variables) {
+    // triggering a new mutation cancels previous queued callbacks
```

Comment:
Yeah, I did this because I was thinking we'd have to manage a `lifecycle` for each trigger of `runMutation`, but I realize now only the most recent trigger should control the `lifecycle` state.

Will fix before merging

Reply:
Added back the `observableQuery` management logic from `Query` and changed subscriptions to a `Set`.

I'm still a bit concerned about possible mutation races (e.g. an call/callback order of `Update(v1), Update(v2) -> Update(v2), Update(v1)` resulting in  `Result(v1)`) but I can't see anything about it in the apollo docs, so maybe it's a non-issue, or something that is best left in userland. If it becomes an issue we can add some `cancelPrevious?` option.